Source/UI/NewUI/BillingForm/Factory/BillingFormFactory.swift

```diff
@@ -0,0 +1,22 @@
+import UIKit
+
+struct BillingFormFactory {
+    // in order
+    static let styles: [BillingFormTextFieldCellStyle] = [
+        DefaultFormStyleFullNameCell(),
+        DefaultFormStyleAddressLine1Cell(),
+        DefaultFormStyleAddressLine2Cell(),
+        DefaultFormStyleCityCell(),
+        DefaultFormStyleStateCell(),
+        DefaultFormStylePostcodedCell(),
+        DefaultFormStyleCountryCell(),
+        DefaultFormStylePhoneNumberCell()
+    ]
+    
+    static func getBillingFormViewController(delegate: BillingFormViewModelDelegate) -> UIViewController {
```

Comment:
Personally I find the passing of a delegate at initialisation time quite uncommon. Not sure this factory provides any value, besides hiding the view model creation, which I am not sure is a common pattern?

Reply:
I understand, what are the common patterns?

Reply:
From my experience, the delegate is commonly exposed as it can occur it may need overriding. But equally its not a requirement to the initialisation of the object, just a requirement for the object to return its updates externally.

Then this factory to create VCs. I don't mind using factories, but I wonder if when you create a VC you **need** to have visibility and control over the requirements at the presentation point. I think the standard flow here when presenting VCs would be

```
let viewModel = ScreenViewModel()
let screen = Screen(with: viewModel)
viewModel.delegate = screen
present(screen)
```

The main benefit for this is the clarity of how the VC and that VM were created and any customisations done at presentation stage. Abstracting externally (don't mind a local method `createScreen`) hides this code that is not really meant to be reused that often (how many places present this screen in the UI flows? )